website/content/docs/connect/config-entries/service-defaults.mdx

```diff
@@ -743,45 +767,49 @@ You can specify the following string values for the `MutualTLSMode` field:
 
 List of extensions to modify Envoy proxy configuration. Refer to [Envoy Extensions](/consul/docs/connect/proxies/envoy-extensions) for additional information.
 
-You can configure the following parameters in the `EnvoyExtensions` block:
+The following table describes how to configure values in the `EnvoyExtensions` map:
 
 | Parameter | Description | Data type | Default |
-| ---   | ---         | ---       | ---     |
-| `Name` | Name of the extension. | string | `""` |
-| `Required` | When Required is true and the extension does not update any Envoy resources, an error is returned. Use this parameter to ensure that extensions required for secure communication are not unintentionally bypassed. | string | `""` |
-| `Arguments` | Arguments to pass to the extension executable. | map | `nil` |
+| ---       | ---         | ---       | ---     |
+| `Name` | Specifies the name of the extension. | String | None |
+| `Required` | Specify `true` to require Consul to implement the extension. <p>Use this parameter to ensure that extensions required for secure communication are not unintentionally bypassed.</p><p>Consul returns an error when `Required` is set to `true` and the extension does not update Envoy resources. </p> | String | None |
+| `Arguments` | Specifies the arguments to pass to the extension executable. Refer to the documentation for the extension you want to implement for additional information. | Map | None |
+| `ConsulVersion` | Specifies the version of Consul that the extension is allowed to work with. Consul validates the version during xDS updates. If a different version is in use, Consul skips the extension and writes the event to the log. <p>The `ConsulVersion` and `EnvoyVersion` must both validate for Consul to implement the extension.</p> | String | None |
+| `EnvoyVersion` | Specifies the version of Envoy that the extension is allowed to work with. Consul validates the version during xDS updates. If a different version is in use, Consul skips the extension and writes the event to the log. <p>The `ConsulVersion` and `EnvoyVersion` must both validate for Consul to implement the extension.</p> | String | None |
```

Comment:
A few suggestions, feel free to edit:
* Linked to `go-version` library for documentation on how to use the version constraint syntax (it can be an interpreted range for flexibility, not just a literal match)
* Clarified how the versions are determined
* Added note on when to use

IMO the last bit about both passing isn't super necessary since it's implied by the failure behavior note (this is just another, optional way for validation/application to "fail").

```suggestion
| `ConsulVersion` | Specifies the Consul [version constraint](https://github.com/hashicorp/go-version) for the extension. Consul validates this against its runtime version during xDS updates. If a non-matching version is in use, Consul will log and skip the extension. <p>Use this parameter to avoid upgrade issues when a configured extension is not compatible with a new version of Consul.</p> | String | None |
| `EnvoyVersion` | Specifies the Envoy [version constraint](https://github.com/hashicorp/go-version) for the extension. Consul validates this against the version of the running Envoy proxy during xDS updates. If a non-matching version is in use, Consul will log and skip the extension. <p>Use this parameter to avoid upgrade issues when a configured extension is not compatible with a new version of Envoy.</p> | String | None |
```